### PR TITLE
[Merged by Bors] - feat: define `MeasureTheory.Lp.const`

### PR DESCRIPTION
**Pull request by urkud**
---

Lean fails to generate the `Norm` instance in 1 lemma. I tried to use `letI` and failed.

- [x] depends on: #5234

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5234
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5234~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by urkud** [comment]
Yes, Lean fails to generate `Norm` instance in the commented lemma.

**Comment by urkud** [comment]
@sgouezel It turns out that the issue was with `NormedField` vs `NontriviallyNormedField`, not with Lean instance search.

File: Mathlib/MeasureTheory/Function/LpSpace.lean [hunk 1]
```diff
@@ -801,6 +808,65 @@ theorem indicatorConstLp_disjoint_union {s t : Set α} (hs : MeasurableSet s) (h
...
+@[simp]
+lemma Memℒp.toLp_const : Memℒp.toLp _ (memℒp_const c) = Lp.const p μ c := rfl
+
+@[simp]
+lemma indicatorConstLp_univ :
+    indicatorConstLp p .univ (measure_ne_top μ _) c = Lp.const p μ c := by
+  rw [← Memℒp.toLp_const, indicatorConstLp]
+  simp only [Set.indicator_univ, Function.const]
+
+theorem Lp.norm_const (hp_zero : p ≠ 0) (hmeas : μ ≠ 0) :
```
Comment (RemyDegenne) [inline Mathlib/MeasureTheory/Function/LpSpace.lean:836]: I think the NeZero PR was merged? If yes, would that be better?
```suggestion
theorem Lp.norm_const (hp_zero : p ≠ 0) [NeZero μ] :
```

**Comment by RemyDegenne** [comment]
LGTM!
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**